### PR TITLE
*DO NOT MERGE* feat(NativeModuleWrapper): Adding native module wrapper concept

### PR DESCRIPTION
**Pull request by rozele**
We can avoid some of the reflection overhead for high throughput modules using the INativeModuleWrapper. This is experimental.

### CODE REVIEW

**Comment by matthargett** [comment]
Is there a sample I can paste into Playground to see the before/after performance of this change?

**Comment by rozele** [comment]
@matthargett good point, I think the best thing might be to try and put a bunch of JS-driven animations in the playground and test before and after FPS

File: ReactWindows/ReactNative.Shared/Modules/Wrappers/UIManagerModuleWrapper.cs [hunk 1]
```diff
@@ -74,100 +74,100 @@ public void OnBatchComplete()
...
-                ToValueArray<int>((JArray) arguments[2]), 
-                ToValueArray<int>((JArray) arguments[3]),
-                ToValueArray<int>((JArray) arguments[4]), 
-                ToValueArray<int>((JArray) arguments[5]));
+                ToInt32(arguments[0]), 
+                ToValueArray(arguments[1], ToInt32),
+                ToValueArray(arguments[2], ToInt32), 
+                ToValueArray(arguments[3], ToInt32),
+                ToValueArray(arguments[4], ToInt32), 
+                ToValueArray(arguments[5], ToInt32));
```
Comment (antonkh) [inline ReactWindows/ReactNative.Shared/Modules/Wrappers/UIManagerModuleWrapper.cs]: These args can be null. Perhaps ToValueArray could just return null in such cases.

**Comment by rozele** [comment]
There is a much better approach to avoiding reflection submitted by @matthargett - #1574

**Pull request closed without merging**